### ISSUE TITLE
Cannot Build Unity Executables on Release 0.14 and 0.14.1 

### ISSUE THREAD

**Issue by sahandrez** [Unity-Technologies/ml-agents#3522]
Hello, 

I have migrated my project from 0.13 to 0.14 and then 0.14.1. The environment works fine in the Editor when I press play but when I want to build the executable, the build will finish with success but with the following errors: 
```
C:\Users\srezaeishoshtari\workspace\UnityProjects\fresh_clone\ml-agents\com.unity.ml-agents\Runtime\Sensor\CameraSensorComponent.cs(9,27): warning CS0109: The member 'CameraSensorComponent.camera' does not hide an accessible member. The new keyword is not required.

C:\Users\srezaeishoshtari\workspace\UnityProjects\fresh_clone\ml-agents\com.unity.ml-agents\Runtime\Policy\BehaviorParameters.cs(34,22): warning CS0649: Field 'BehaviorParameters.m_BehaviorType' is never assigned to, and will always have its default value 

Build completed with a result of 'Succeeded'
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr) (at C:/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:179)
```
And when I execute the file, a broken empty environment starts (see attached image) starts.
At first, I thought the issue was with the migration of our project, but then I did a fresh clone of ml-agents (tried with both releases 0.14 and 0.14.1) and the exact same thing happens with the provided example environments. 

This is the log file of when I launch the .exe file:
```
Mono path[0] = 'C:/Users/srezaeishoshtari/workspace/UnityProjects/fresh_clone/built/test/Unity Environment_Data/Managed'
Mono config path = 'C:/Users/srezaeishoshtari/workspace/UnityProjects/fresh_clone/built/test/MonoBleedingEdge/etc'
PlayerConnection initialized from C:/Users/srezaeishoshtari/workspace/UnityProjects/fresh_clone/built/test/Unity Environment_Data (debug = 0)
PlayerConnection initialized network socket : 0.0.0.0 55204
Multi-casting "[IP] 10.128.12.182 [Port] 55204 [Flags] 2 [Guid] 2354467184 [EditorId] 2630875911 [Version] 1048832 [Id] WindowsPlayer(MTL-BG699) [Debug] 0 [PackageName] WindowsPlayer" to [225.0.0.222:54997]...
Started listening to [0.0.0.0:55204]
PlayerConnection already initialized - listening to [0.0.0.0:55204]
Initialize engine version: 2018.4.17f1 (b830f56f42f0)
GfxDevice: creating device client; threaded=1
Direct3D:
    Version:  Direct3D 11.0 [level 11.1]
    Renderer: NVIDIA GeForce GTX 1070 (ID=0x1b81)
    Vendor:   
    VRAM:     8088 MB
    Driver:   25.21.14.2600
Begin MonoManager ReloadAssembly
- Completed reload, in  0.731 seconds
<RI> Initializing input.

<RI> Input initialized.

<RI> Initialized touch support.

The referenced script (SerializeAgent) on this Behaviour is missing!
 
(Filename: C:\buildslave\unity\build\Runtime/Scripting/ManagedReference/SerializableManagedRef.cpp Line: 195)

The referenced script on this Behaviour (Game Object 'Agent') is missing!
 
(Filename: C:\buildslave\unity\build\Runtime/Mono/ManagedMonoBehaviourRef.cpp Line: 294)

A scripted object (probably SerializeAgent?) has a different serialization layout when loading. (Read 32 bytes but expected 44 bytes)
Did you #ifdef UNITY_EDITOR a section of your serialized properties in any of your scripts?
 
(Filename: C:\buildslave\unity\build\Runtime/Serialize/SerializedFile.cpp Line: 2002)

UnloadTime: 0.719300 ms
Setting up 6 worker threads for Enlighten.
  Thread -> id: 2aec -> priority: 1 
  Thread -> id: 3e90 -> priority: 1 
  Thread -> id: 85f4 -> priority: 1 
  Thread -> id: 9314 -> priority: 1 
  Thread -> id: 88f4 -> priority: 1 
  Thread -> id: 8e14 -> priority: 1 
```
Two files are missing which seem might be due to some Windows/Mac compatibility. The first half of the path to the file is in Windows format and the rest Unix. 

I have tested on Unity 2018.4.17f1 and 2019.3.3f1. I never had such issues with ML-Agents 0.13. This seems like a bug to me or else I am missing some thing. 

Thanks, 
Sahand

**Comment by sahandrez** [comment]
This is a screenshot of the environment. 

![unityexe](https://user-images.githubusercontent.com/32560115/75486786-2e3fbd00-597b-11ea-999c-29e7f9526952.png)

**Comment by surfnerd** [comment]
Hi @sahandrez,
In the build dialog you need to add the open scene and remove this one.  I'll fix this for the next release.

**Comment by surfnerd** [comment]
Fix is here: #3524
